src/Geometry/CTransform.cs

```diff
@@ -76,17 +76,52 @@ private static double[][] GetUniformMatrix(double value)
 			return matrix;
 		}
 
+		/// <summary>Returns a Identity Matrix</summary>
+		private static double[,] GetIdentityMatrix()
+		{
+			double[,] matrix =
+			{
+				{ 1, 0, 0, 0 }, { 0, 1, 0, 0 },
+				{ 0, 0, 1, 0 }, { 0, 0, 0, 1 }
+			};
+			return matrix;
+		}
+
 		/// <summary>Returns the value at the given coordinate</summary>
 		public double this[int row, int column]
 		{
-			get => _transforms[row][column];
-			set => _transforms[row][column] = value;
+			get => _transforms[row,column];
+			set => _transforms[row, column] = value;
+		}
+
+		public static double[] Flatten(double[,] twoDimensionalArray)
+		{
+			int numRows = twoDimensionalArray.GetLength(0);
+			int numCols = twoDimensionalArray.GetLength(1);
+			double[] oneDimensionalArray = new double[numRows * numCols];
+			int index = 0;
+
+			for (int row = 0; row < numRows; row++)
+			{
+				for (int col = 0; col < numCols; col++)
+				{
+					oneDimensionalArray[index] = twoDimensionalArray[row, col];
+					index++;
+				}
+			}
+
+			return oneDimensionalArray;
 		}
 
 		/// <summary>Tests the Matrix for any NaN's or infinity numbers</summary>
 		public bool IsValid()
 		{
-			double[] values = _transforms.SelectMany(t => t).ToArray();
+			double[] values = Flatten(_transforms);
+
+			// Check that the matrix is all zeros
+			if (values.All(v => v == 0))
+				return true;
```

Comment:
Should this be valid? If so I need to update my check in Crash!
And if it is valid, is there a way to check for "IsDefault" or something? This transform in Rhino turns things inside out somehow and it'd be great if I can discard 0 transforms :)

Reply:
Maybe `IsIdentity?` Just saw the method above

Reply:
everything 0 should actually not be valid. The identity matrix means nothing happens. The 0 matrix is not valid because if we add any transformation to it the product of it will also result to 0. Its kind of multiplying all the transformations with 0 which results to nothing